pkg/volume/csi/csi_attacher.go

```diff
@@ -106,6 +107,12 @@ func (c *csiAttacher) Attach(spec *volume.Spec, nodeName types.NodeName) (string
 		},
 	}
 
+	// Begin watching for this volume attachment prior to creating it to avoid race conditions.
+	if c.plugin.volumeAttachmentInformerHook != nil {
+		_ = c.plugin.volumeAttachmentInformerHook.CreateWatch(attachID, watchChannelBufferSize)
+		defer c.plugin.volumeAttachmentInformerHook.DeleteWatch(attachID)
+	}
+
 	_, err = c.k8s.StorageV1().VolumeAttachments().Create(context.TODO(), attachment, metav1.CreateOptions{})
```

Comment:
I was thinking with sharedinformer, create/delete/update/list can all be used with informer instead of directly updating api server?

Reply:
I don't think there's a way to make those API calls directly through the SharedInformer.

Reply:
sorry my question is not correct. based on https://github.com/kubernetes/kubernetes/issues/64429, before creating/deleting, could use informer (lister) to check whether object exists or not to save some unnecessary calls.

Reply:
Oh I see what you mean. I had previously tried doing this in this PR but it introduced some complexity that I wasn't sure how to simplify.

I have some remnants of this approach [in this branch](https://github.com/chrishenzie/kubernetes/compare/volume-attachment-informer...volume-attachment-informer-tip). In order to make this work, we need a way of ensuring an object created through the API is available through the lister. One way we could do this is by prepending a reactor that updates the informer immediately, but we need to plumb that reactor function through to the CSI test plugin constructor (see the `updateInformerReactionFunc` type and changes to `newTestPluginWithVolumeHost()`). I can play around with this again today and see if I can get it working.